apps/passport-server/migrations/61_rate_limit_variable_periods.sql

```diff
@@ -0,0 +1,63 @@
+-- Moving rate limit configuration to TypeScript code, so this is no longer
```

Comment:
Now that we're re-implementing this rather than borrowing an off-the-shelf solution, I wonder if there's a strong need for doing this in a SQL function rather than Typescript.  I can see a performance argument (fewer round-trips to the database), but it's offset by my worry that I don't know enough about SQL functions to review this properly.  While we do have some decent unit tests, I don't trust them to be able to fully cover all the edge-cases and race-conditions at this level.

Reply:
Clarifying after discussion on Discord: This comment is about the SQL function not the SQL table.  It's the logic in SQL which makes me nervous.  The state needs to be in SQL to persist against restarts and (eventually) be shared between multiple server instances.

This isn't a strong suggestion, just expressing that I'm nervous, since I'm not familiar enough with the tradeoffs and risks of SQL functions to be sure what's safe.  That said I realize that some tricky complexity around SQL is needed regardless, e.g. in the form of locks to ensure the logic is safely atomic.